### PR TITLE
Hazardous Gases (Cl and HCl)

### PR DESCRIPTION
**Pull request by Thera-Pissed**
## About The Pull Request
Adds chlorine gas and hydrogen chloride gas
These gases hurt to be exposed to in high concentration if not in EVA
Plasma, Trit, Cl, and HCl now damage eyes if not wearing goggles (welding, heat, or cold goggles currently)
Chlorine reacts with hydrogen to make HCl and makes heat while doing so. HCl burns into H2O and Cl.
image
image

Scrubbers are now set to filter most gases at roundstart.

## Why It's Good For The Game
Hazardous planetary gases! (not actually implemented in this PR, but opens up the possibility)
More gases that are actually relevant to Normal shiptest rounds is good, since there's few station gases that can be reasonably seen or made (freon has been made like once. stimulum and nitryl are dumb. likely could be removed)

## Changelog

:cl:
add: Chlorine and Hydrogen Chloride gases! Wear your goggles!
/:cl:

### CODE REVIEW

**Comment by Thera-Pissed** [comment]
woohoo 164 commits because I started before hydrogen got in!

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -54,7 +54,9 @@
 		"caution" = /obj/machinery/portable_atmospherics/canister,
 		"freon" = /obj/machinery/portable_atmospherics/canister/freon,
 		"hydrogen" = /obj/machinery/portable_atmospherics/canister/hydrogen,
-		"fuel mix" = /obj/machinery/portable_atmospherics/canister/fuel
+		"fuel mix" = /obj/machinery/portable_atmospherics/canister/fuel,
+		"chlorine" = /obj/machinery/portable_atmospherics/canister/chlorine,
+		"hydrogen chloride" =/obj/machinery/portable_atmospherics/canister/hydrogen_chloride
 	)
```
Comment (rye-rice) [inline code/modules/atmospherics/machinery/portable/canister.dm]: o,
```suggestion
		"fuel mix" = /obj/machinery/portable_atmospherics/canister/fuel,
		"chlorine" = /obj/machinery/portable_atmospherics/canister/chlorine,
		"hydrogen chloride" =/obj/machinery/portable_atmospherics/canister/hydrogen_chloride,
	)
```

**Review by rye-rice — COMMENTED** [review]
Push a new commit, seems checks failed for some unrelated reason

**Comment by rye-rice** [comment]
oh wow that fixed the checks

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -61,6 +61,21 @@
 			return ONE_ATMOSPHERE
 	return pressure
 
+/mob/living/carbon/human/proc/check_for_seal()
+	if (wear_suit && head && istype(wear_suit, /obj/item/clothing) && istype(head, /obj/item/clothing))
+		var/obj/item/clothing/CS = wear_suit
+		var/obj/item/clothing/CH = head
+		if (CS.clothing_flags & CH.clothing_flags & STOPSPRESSUREDAMAGE)
+			return TRUE
+	return FALSE
```
Comment (SomeguyManperson) [inline code/modules/mob/living/carbon/human/life.dm]:
```suggestion
	var/obj/item/clothing/CS = wear_suit
	var/obj/item/clothing/CH = head
	if(istype(CS) && istype(CH))
		if (CS.clothing_flags & CH.clothing_flags & STOPSPRESSUREDAMAGE)
			return TRUE
	return FALSE
```

**Comment by rye-rice** [comment]
Tell me if this still isn't finished, I'll merge in like 2 days after skimming the code if a review isn't given

**Comment by firebudgy** [comment]
Spilled a bottle of chlorine on a local and it runtimed.
image
image

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -649,6 +652,33 @@
...
 		mytray.adjustWeeds(-rand(1,3))
 
+/datum/reagent/chlorine/expose_obj(obj/O, reac_volume)
+	if((!O) || (!reac_volume))
+		return 0
+	var/temp = holder ? holder.chem_temp : T20C
+	O.atmos_spawn_air("chlorine=[reac_volume/2];TEMP=[temp]")
+
+/datum/reagent/chlorine/expose_turf(turf/open/T, reac_volume)
+	if(istype(T))
```
Comment (rye-rice) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: Avoid single letter vars, changing obj/O to obj/exposed_object shouldn't mess anything up i think, same with turf/open/T

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -649,6 +652,33 @@
...
+	color = "#f4ffe0"
+	taste_description = "acid"
+
+/datum/reagent/chlorine/on_mob_life(mob/living/carbon/M)
+	M.take_bodypart_damage(0, 2*REM, 0, 0)
+	M.adjustOrganLoss(ORGAN_SLOT_LUNGS,1*REM)
+	M.adjustOrganLoss(ORGAN_SLOT_STOMACH,1*REM)
+	. = 1
+	..()
+
```
Comment (rye-rice) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: Same here with M

**Review by rye-rice — CHANGES_REQUESTED** [review]

**Review by rye-rice — APPROVED** [review]

**Pull request merged**